src/main/java/InfrastructureManager/Rest/RestOutput.java

```diff
@@ -87,4 +96,18 @@ public void resetOutput() {
         this.output = new LinkedList<>();
         this.limitNodes = new LinkedHashMap<>();
     }
+
+    private void createNodeResource(String clientID, String nodeID) {
+        try {
+            EdgeClient client = Master.getInstance().getClientByID(clientID);
+            EdgeNode node = Master.getInstance().getNodeByID(nodeID);
+            if (node == null) {
+                throw new Exception("Client not found");
```

Comment:
is that a client or a node? why not for both?

Reply:
My bad, this is from the old implementation. This is not even needed as both methods `getClientByID` and `getNodeByID` cannot return null and already throw an exception when the resource is not found. The last commit already has this fixed.